packages/ui/src/forum/modals/CreateThreadModal/machine.ts

```diff
@@ -0,0 +1,57 @@
+import { createMachine, assign } from 'xstate'
+
+import { EmptyObject } from '@/common/types'
+
+export interface GeneralFormContext {
+  topic?: string
+  description?: string
+}
+
+type TransactionContext = Required<GeneralFormContext>
+
+type CreateThreadContext = Partial<TransactionContext>
+
+type CreateThreadState =
+  | { value: 'requirementsVerification'; context: EmptyObject }
+  | { value: 'requirementsFailed'; context: EmptyObject }
+  | { value: 'generalDetails'; context: GeneralFormContext }
+  | { value: 'end'; context: TransactionContext }
```

Comment:
One last thing, I think that one or two of those types are redundant here:

```suggestion
export interface CreateThreadContext {
  topic?: string
  description?: string
}

type CreateThreadState =
  | { value: 'requirementsVerification'; context: EmptyObject }
  | { value: 'requirementsFailed'; context: EmptyObject }
  | { value: 'generalDetails'; context: CreateThreadContext }
  | { value: 'end'; context: Required<CreateThreadContext> }
```

The `Required<CreateThreadContext>` can be brought back as well, but since it will be a very small machine we can leave that as inline type.